xlgui/guiutil.py

```diff
@@ -310,6 +313,17 @@ def __init__(self, entry=None, timeout=500):
         entry.connect('icon-press', self.on_entry_icon_press)
         entry.connect('activate', self.on_entry_activated)
 
+        # Disable global accelerators while editing, re-enable when
+        # done
+        entry.connect(
+            'focus-in-event',
+            lambda w, e: xlgui.get_controller().main.accel_manager.disable_accelerators(),
```

Comment:
This seems fine to me. The only question I have is whether there's a guarantee that the `focus-out-event` will always be emitted? Though, now that I've typed it, that seems silly that it wouldn't be guaranteed.